apis/gateway/v1beta2/apirule_conversion_test.go

```diff
@@ -0,0 +1,516 @@
+					Rules: []v1beta2.Rule{
+						{
+							Jwt: &v1beta2.JwtConfig{
+								Authentications: []*v1beta2.JwtAuthentication{
+									{
+										Issuer:      "issuer",
+										JwksUri:     "jwksUri",
+										FromHeaders: jwtHeadersBeta2,
+										FromParams:  []string{"param1", "param2"},
+									},
+								},
+								Authorizations: []*v1beta2.JwtAuthorization{
+									{
+										RequiredScopes: []string{"scope1", "scope2"},
+										Audiences:      []string{"audience1", "audience2"},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			apiRuleBeta1 := v1beta1.APIRule{}
+
+			// when
+			err := apiRuleBeta2.ConvertTo(&apiRuleBeta1)
+
+			// then
+			Expect(err).To(BeNil())
+			Expect(apiRuleBeta1.Spec.Rules).To(HaveLen(1))
+			Expect(apiRuleBeta1.Spec.Rules[0].AccessStrategies).To(HaveLen(1))
+			Expect(apiRuleBeta1.Spec.Rules[0].AccessStrategies[0].Handler.Name).To(Equal("jwt"))
+			Expect(apiRuleBeta1.Spec.Rules[0].AccessStrategies[0].Config).ToNot(BeNil())
+
+			jwtConfigBeta1 := apiRuleBeta1.Spec.Rules[0].AccessStrategies[0].Config.Object.(*v1beta2.JwtConfig)
+
+			Expect(jwtConfigBeta1.Authentications).To(HaveLen(1))
+			Expect(jwtConfigBeta1.Authentications[0].Issuer).To(Equal("issuer"))
+			Expect(jwtConfigBeta1.Authentications[0].JwksUri).To(Equal("jwksUri"))
+			Expect(jwtConfigBeta1.Authentications[0].FromHeaders).To(HaveLen(1))
+			Expect(jwtConfigBeta1.Authentications[0].FromHeaders[0].Name).To(Equal(jwtHeadersBeta2[0].Name))
+			Expect(jwtConfigBeta1.Authentications[0].FromHeaders[0].Prefix).To(Equal(jwtHeadersBeta2[0].Prefix))
+			Expect(jwtConfigBeta1.Authentications[0].FromParams).To(HaveExactElements("param1", "param2"))
+			Expect(jwtConfigBeta1.Authorizations).To(HaveLen(1))
+			Expect(jwtConfigBeta1.Authorizations[0].RequiredScopes).To(HaveExactElements("scope1", "scope2"))
+			Expect(jwtConfigBeta1.Authorizations[0].Audiences).To(HaveExactElements("audience1", "audience2"))
+		})
+
+		It("should convert single rule with JWT and ignore NoAuth when set to false to v1beta1", func() {
+			// given
+			apiRuleBeta2 := v1beta2.APIRule{
+				Spec: v1beta2.APIRuleSpec{
+					Hosts: []*string{&host1},
+					Rules: []v1beta2.Rule{
+						{
+							NoAuth: ptr.To(false),
+							Jwt: &v1beta2.JwtConfig{
+								Authentications: []*v1beta2.JwtAuthentication{},
+								Authorizations:  []*v1beta2.JwtAuthorization{},
+							},
+						},
+					},
+				},
+			}
+			apiRuleBeta1 := v1beta1.APIRule{}
+
+			// when
+			err := apiRuleBeta2.ConvertTo(&apiRuleBeta1)
+
+			// then
+			Expect(err).To(BeNil())
+			Expect(apiRuleBeta1.Spec.Rules).To(HaveLen(1))
+			Expect(apiRuleBeta1.Spec.Rules[0].AccessStrategies).To(HaveLen(1))
+			Expect(apiRuleBeta1.Spec.Rules[0].AccessStrategies[0].Handler.Name).To(Equal("jwt"))
+			Expect(apiRuleBeta1.Spec.Rules[0].AccessStrategies[0].Config).ToNot(BeNil())
+		})
+
+		It("should convert two rules with NoAuth and JWT to v1beta1", func() {
```

Comment:
Still we are using an invalid v1beta2 as test data, I find it confusing. The tests are also a documentation of functionality. I would avoid this confusion and use only valid cases for tests. We can just have two rules and it would be the same test, right?